src/placement/jump_map.c

```diff
@@ -519,12 +519,6 @@ obj_remap_shards(struct pl_jump_map *jmap, struct daos_obj_md *md,
 			get_target(root, &spare_tgt, crc(key, rebuild_key),
 				   dom_used, tgts_used, shard_id, op_type);
 			spares_left--;
-			if (can_extend(op_type, spare_tgt->ta_comp.co_status)) {
-				rc = remap_alloc_one(extend_list, shard_id,
-						spare_tgt, true);
-				if (rc)
-					return rc;
-			}
```

Comment:
We don't currently have any tests that cover this, but this seems like it was probably added here for a reason.... I suspect removing this code is going to break addition somehow. Why was this removed?

Reply:
For example, for shard A, the rebuild sequence with the JCH are

tgt22 ---> tgt17--->tgt25  (suppose tgt22 and tgt 17 are all in rank 2). i.e. if we kill rank2,  tgt22 will be rebuilt on tgt25 anyway, because tgt17 is also failed.

Then reintegrate rank2,  it needs move data back directly from tgt25 --> tgt22 instead of through tgt17.

During the remap,  initially it has tgt22 then it get tgt17 due to JCH, but tgt17 is also in UP status, so it will go further to find out tgt25.  But we should not add tgt 17 to the extending list,  because there won't be any data to be reintegrated on tgt17 and also this will break the object layout with inflight IO, and causing that assertion.

I assume the initial implementation is just trying follow the rebuild remap list, though this is not quite right for extending list.

Reply:
in above example, tgt22 ---> tgt17--->tgt25 is the spare tgt selection list? why the two tgts (22 and 17) on same node can be adjacent to each other, in the first place?